### ISSUE TITLE
Google removed the extension from the Chrome webstore

### ISSUE THREAD

**Issue by eddex** [eddex/hslu-simple-mep-results#98]
The extension can't be found on the Chrome webstore anymore.

https://chrome.google.com/webstore/detail/hslu-simple-mep-results/bkcgbpgefenaapagldnnabfakilmfihp leads to a 404.

RIP

**Comment by simonbaumeler** [comment]
Isn't there a chrome webstore account that should have been notified what happened with the extension and how to get it back online?

**Comment by Lextum** [comment]
@simonbaumeler Yes there is. We reason was the permission  `"webRequest"` we apparently don't use. I migrated to v3 and removed the permission. But I can't test if it still works. Could you check out the branch `manifest_v3` and try it out? Please tell me, if you need help installing the extension directly from source.

**Comment by Lextum** [comment]
@Lextum We definitely need to give this project to STAIR or have some maintainers that have access to MyCampus.

**Comment by simonbaumeler** [comment]
doesnt work yet:

{
  "manifest_version": 3,
  "name": "HSLU simple MEP results",
  "version": "2.1.1",
  "description": "Show the results of your exams in a nice way instead of the crappy default layout.",
  "icons": {
    "48": "icons/icon-48.png",
    "96": "icons/icon-96.png",
    "128": "icons/icon-128.png"
  },
  "action": {
    "default_icon": "icons/icon-48.png",
    "default_title": "HSLU simple MEP results",
    "default_popup": "popup/popup.html",
    "browser_style": true
  },
  "content_scripts": [
    {
      "matches": [
        "://.mycampus.hslu.ch/*/stud-i/mein-studium/meine-anmeldungen/*"
      ],
      "js": [
        "components/helpers.js",
        "components/i18n.js",
        "components/module_parser.js",
        "lib/Chart.bundle.min.js",
        "main.js"
      ],
      "css": [
        "templates/custom_styles.css"
      ]
    }
  ],
  "permissions": [
    "storage"
  ],
  "host_permissions": [
    "https://mycampus.hslu.ch/*"
  ],
  "web_accessible_resources": [
    {
      "resources": [
        "templates/grades_table.html",
        "templates/credits_by_module_type_table.html",
        "templates/custom_styles.css",
        "data/modules_i.json",
        "data/modules_ics.json",
        "data/modules_wi.json",
        "data/modules_ai.json",
        "_locales/*"
      ],
      "matches": [],
      "extension_ids": []
    }
  ],
  "default_locale": "de",
  "browser_specific_settings": {
    "gecko": {
      "id": "{64b1a494-88df-447e-b2aa-45d86fcc3c16}",
      "update_url": "https://raw.githubusercontent.com/eddex/hslu-simple-mep-results/master/updates.json"
    }
  }
}

![WhatsApp Image 2022-10-20 at 19 23 31](https://user-images.githubusercontent.com/26277542/197017121-9adc17d9-cac3-4cb9-b48f-c9a94a9e51c0.jpeg)

**Comment by Lextum** [comment]
I believe you can ignore this error. The extension should still load inside MyCampus.

**Comment by simonbaumeler** [comment]
The extension does not load when opening the "Meine Anmeldungen" site.

**Comment by Lextum** [comment]
Can you remove this part from the manifest and try again?

**Comment by simonbaumeler** [comment]
I'll try tomorrow 👍🏼

**Comment by simonbaumeler** [comment]
i removed the part from the manifest and got no error to start with afterwards but when loading the "meine Anmeldungen" page the extension still did not load. 

Extension error:
![image](https://user-images.githubusercontent.com/26277542/197178128-b1b4392d-ff55-4e51-a0fd-a10d18abb5f7.png)

in text:
/**
 * Returns the default locale from the manifest
 * @returns {string}
 */
function getDefaultLocale() {
    if (Helpers.isFirefox()) {
        return (browser.runtime.getManifest()).default_locale
    }
    else {
        return (chrome.runtime.getManifest()).default_locale
    }
}

/**
 * Returns the language of MyCampus
 * @returns {string}
 */
async function getLanguage() {
    languageLinks = document.getElementsByClassName("languagelink");
    if (languageLinks.length > 0) {
        for (let i = 0; i < languageLinks.length; i++) {
            const languageLink = languageLinks[i];
            if (languageLink.classList.contains("active")) {
                return languageLink.hreflang;
            }
        }
    }
    // needed for the popup
    else {
        const i18nLanguage = (await Helpers.getItemFromLocalStorage("i18nLanguage")).i18nLanguage
        if (i18nLanguage === undefined) {
            return getDefaultLocale();
        }
        return i18nLanguage
    }
}

/**
 * 
 * @param {*} language 
 * @returns a object with all i18n strings for the given language
 */
async function getMessages(language) {
    let messsages = await fetch(Helpers.getExtensionInternalFileUrl('_locales/' + language + '/messages.json'))
        .then(response => response.json())
    if (messsages === undefined) {
        messsages = await fetch(Helpers.getExtensionInternalFileUrl('_locales/' + getDefaultLocale() + '/messages.json'))
            .then(response => response.json())
    }
    return messsages;
}

/**
 * Component for localization functions.
 */
const i18n = {

    messages: "",
    /**
     * Gets the right language and the corresponding message file
     */
    init: async () => {

        const language = await getLanguage();
        i18n.messages = await getMessages(language);
        await Helpers.saveObjectInLocalStorage({ i18nLanguage: language })
    },

    /**
     * Returns the localized message
     * @param {string} message wo should be localized
     */
    getMessage: (message) => {
        if (message in i18n.messages) {
            return i18n.messages[message].message
        }
        return undefined
    },

}

Console errors:
![image](https://user-images.githubusercontent.com/26277542/197178186-947f60a5-1d2e-4986-8cc4-31f894e28e13.png)

**Comment by simonbaumeler** [comment]
hello @Lextum 

I hope you are well. How is the status here?

**Comment by Lextum** [comment]
Hello @simonbaumeler 

Sorry, I completely forget about this issue. I contacted the Google Support and tried to resubmit the extension but they insisted on removing the WebRequest permission. Neither @eddex nor myself have access to MyCampus, so we can't test or try to find a solution for this problem. I you want to help us find a solution or can prove somehow that this permission is needed, I am happy to reupload the extension.

**Comment by simonbaumeler** [comment]
Hello @Lextum 

Thank you for your Feedback, I've reached out to STAIR in order to organise something to keep this nice extension alive and up-to-date. Also if you tell me what to check for i can help you find a solution or we could even have a session on Discord in which i could show you everything you need on mycampus.

**Comment by Lextum** [comment]
Hi @simonbaumeler 

Sure, we can try to find a solution together. Just send me a PM on discord at Letum#5528. If you want to start some research, I would say the error you got in an answer above is a good starting point.
